### DISCUSSION TITLE
Roadmap for ray-optics

### DISCUSSION THREAD

**Discussion by dibyendumajumdar** [mjhoptics/ray-optics#36] [General]
It will be good to know what the development roadmap is for ray-optics. 

I would like to see support for geometric MTF curves.

**Comment by mjhoptics** [comment]
The closest thing to a roadmap I can offer is this in the [Introduction](https://ray-optics.readthedocs.io/en/latest/Introduction.html). I'm retired and one of the things I retired from was roadmaps ;-). My current interest is in GUI tools for first order optical design. This is an area that hasn't been well explored, IMHO, and is an area I've long been interested in. 

I'm trying to structure the capabilities in ray-optics so that basic calculations are provided and available to build more sophisticated analyses on top. The MTF is the Fourier transform of the PSF, but it gets more complicated when you 1) want the polychromatic MTF and 2) systems are fast and well corrected, requiring the effects of diffraction to be taken into account. So I see it as a non-trivial project.

**Reply by dibyendumajumdar** [reply]
> The closest thing to a roadmap I can offer is this in the [Introduction](https://ray-optics.readthedocs.io/en/latest/Introduction.html). I'm retired and one of the things I retired from was roadmaps ;-). 

Of course! I am envious you are able to choose what to work on! 

> My current interest is in GUI tools for first order optical design. This is an area that hasn't been well explored, IMHO, and is an area I've long been interested in. 

Understood. 

> I'm trying to structure the capabilities in ray-optics so that basic calculations are provided and available to build more sophisticated analyses on top. The MTF is the Fourier transform of the PSF, but it gets more complicated when you 1) want the polychromatic MTF and 2) systems are fast and well corrected, requiring the effects of diffraction to be taken into account. So I see it as a non-trivial project.

I wish I could say I will contribute this but I have no idea how to implement it. Maybe one day!
